lib/spack/spack/cmd/buildcache.py

```diff
@@ -120,26 +108,27 @@ def find_matching_specs(specs, allow_multiple_matches=False, force=False):
     # List of specs that match expressions given via command line
     specs_from_cli = []
     has_errors = False
-    for spec in specs:
-        matching = spack.store.db.query(spec)
-        # For each spec provided, make sure it refers to only one package.
-        # Fail and ask user to be unambiguous if it doesn't
-        if not allow_multiple_matches and len(matching) > 1:
-            tty.error('{0} matches multiple packages:'.format(spec))
-            print()
-            spack.cmd.display_specs(matching, **display_args)
-            print()
-            has_errors = True
-
-        # No installed package matches the query
-        if len(matching) == 0 and spec is not any:
-            tty.error('{0} does not match any installed packages.'.format(
-                spec))
-            has_errors = True
-
-        specs_from_cli.extend(matching)
-    if has_errors:
-        tty.die(error_message)
+    for pkg in pkgs:
```

Comment:
This outer for loop (`for pkg in pkgs`) should not be required. You should be able to obtain all specs by calling `spack.cmd.parse_specs` on `pkgs` (although you would have to undo the code where you construct `pkgs` as a set - you would want to use it as just a list). `cmd.install` has an example of parsing out multiple specs from the `pkgs` argument. Let me know if that doesn't make sense.

Reply:
Wouldn't parse_specs concretize based on the current set of package.py files? I was trying to match the specs as downloaded. I do use parse_specs when looking for installed packages to make buildcaches.

Reply:
Er... now that I see that in context it makes more sense.